src/vs/workbench/contrib/terminal/browser/terminalTooltip.ts

```diff
@@ -6,33 +6,25 @@
 import { localize } from 'vs/nls';
 import { ITerminalInstance } from 'vs/workbench/contrib/terminal/browser/terminal';
 import { TerminalCapability } from 'vs/platform/terminal/common/capabilities/capabilities';
+import { IConfigurationService } from 'vs/platform/configuration/common/configuration';
+import { TerminalSettingId } from 'vs/platform/terminal/common/terminal';
 
-function getCapabilityName(capability: TerminalCapability): string | undefined {
-	switch (capability) {
-		case TerminalCapability.CwdDetection:
-		case TerminalCapability.NaiveCwdDetection:
-			return localize('capability.cwdDetection', "Current working directory detection");
-		case TerminalCapability.CommandDetection:
-			return localize('capability.commandDetection', "Command detection");
-		case TerminalCapability.PartialCommandDetection:
-			return localize('capability.partialCommandDetection', "Command detection (partial)");
+export function getShellIntegrationTooltip(instance: ITerminalInstance, markdown: boolean, configurationService: IConfigurationService): string {
+	if (!configurationService.getValue(TerminalSettingId.ShellIntegrationEnabled)) {
+		return '';
 	}
-}
-
-export function getShellIntegrationTooltip(instance: ITerminalInstance, markdown?: boolean): string {
-	let shellIntegrationString = '';
 	const shellIntegrationCapabilities: TerminalCapability[] = [];
 	if (instance.capabilities.has(TerminalCapability.CommandDetection)) {
 		shellIntegrationCapabilities.push(TerminalCapability.CommandDetection);
 	}
 	if (instance.capabilities.has(TerminalCapability.CwdDetection)) {
 		shellIntegrationCapabilities.push(TerminalCapability.CwdDetection);
 	}
+	let shellIntegrationString = '';
 	if (shellIntegrationCapabilities.length > 0) {
-		shellIntegrationString += `${markdown ? '\n\n---\n\n' : '\n\n'} ${localize('shellIntegration.enabled', "Shell integration is enabled")}`;
-		for (const capability of shellIntegrationCapabilities) {
-			shellIntegrationString += `\n- ${getCapabilityName(capability)}`;
-		}
+		shellIntegrationString += `${markdown ? '\n\n---\n\n' : '\n\n'} ${localize('shellIntegration.enabled', "Shell integration activated")}`;
+	} else {
+		shellIntegrationString += `${markdown ? '\n\n---\n\n' : '\n\n'} ${localize('shellIntegration.activationFailed', "Shell integration failed to activate")}`;
```

Comment:
This is technically "did not activate", should we say that until we have a time based mechanism that expected it to activate after a few seconds?

Reply:
Well "did not activate" can happen because it is not enabled with the setting. In this case, we tried to activate it and it failed.

Reply:
But it hasn't necessarily failed yet, consider the case of creating a shell that takes a few seconds to load and you hover it immediately. Small thing and I guess we could improve as part of https://github.com/microsoft/vscode/issues/148840 since we'll do the timeout thing there?